### PR TITLE
Tooling & test improvements

### PR DESCRIPTION
**Pull request by usirin**
### npm run format:single <filepath>

will format a single file with given path using `prettier` and default config.

### npm run format:all

will format all `js` files under `src` folder.

### npm run format:check:single <filepath>

will check if given file is formatted correctly, if not it's gonna exit with a non-zero exit code.

### npm run format:check:all

will check if all `js` files are formatted correctly, if not it's gonna exit with a non-zero exit code. (Gonna be used by CI environment[1] to make sure all files are still formatted correctly)

### npm run test

will now check all files matching `src/**/**.test.js` effectively allowing us to co-exist tests with the related module[2].

[1]: `travis` will run `npm run format:check:all` along with `npm test`.
[2]: it was matching `test/**/*.js` before.

### CODE REVIEW

**Comment by usirin** [comment]
@gokmen without porting `kite-server.coffee` it looks really hard to get the old tests running again. If we want this, i can work on that. But instead of doing that, i think working on unit tests of kite itself would be more useful at this point. There are really little amount of tests, but the implementation is complex enough to make them less useful actually.

Let's talk about this.

Other than that, i think this pr is complete as a structural change. i will open other prs for individual tests.

File: src/kite/base.js [hunk 1]
```diff
@@ -177,20 +183,25 @@ module.exports = Kite = (() => {
...
 
       this.proto.emit('request', scrubbed)
     }
 
-    expireTokenOnExpiry () {
-      if ((this.options.auth != null ? this.options.auth.type : undefined) !== 'token') {
+    expireTokenOnExpiry() {
+      if (
+        (this.options.auth != null ? this.options.auth.type : undefined) !==
+        'token'
```
Comment (gokmen) [inline src/kite/base.js]: this looks weird.

File: src/kite/delayed.js [hunk 2]
```diff
@@ -14,15 +14,15 @@ module.exports = Delayed = (() => {
       this.prototype.clear = mustOverride
     }
 
-    constructor (fn, ms, ...rest) {
+    constructor(fn, ms, ...params) {
       this.fn = fn
-      this.ms = ms;
-      [...this.params] = Array.from(rest)
+      this.ms = ms
+      this.params = params
```
Comment (gokmen) [inline src/kite/delayed.js]: will it be a copy of `params` or should we keep `Array.from` ?

File: src/kite/enableLogging.js [hunk 3]
```diff
@@ -1,11 +1,19 @@
-const [CRITICAL, ERROR, WARNING, NOTICE, INFO, DEBUG] = Array.from([0, 1, 2, 3, 4, 5])
+const [CRITICAL, ERROR, WARNING, NOTICE, INFO, DEBUG] = Array.from([
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+])
```
Comment (gokmen) [inline src/kite/enableLogging.js]: how about `Array.from(Array(6).keys())`

File: src/kontrol/index.js [hunk 4]
```diff
@@ -3,12 +3,24 @@ const BaseKontrol = require('./base')
...
+      this.Kite = Kite
+      methods.forEach(
+        method =>
+          (Kontrol.prototype[method] = Promise.promisify(
+            BaseKontrol.prototype[method]
+          ))
+      )
     }
   }
   Kontrol.initClass()
```
Comment (gokmen) [inline src/kontrol/index.js]: we need to remove generated `initClass` helpers at some point.

File: src/server/index.test.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      autoConnect: false,
+    })
+
+    const server = new Server({ port: 7777 })
+    server.on('connection', connection => {
+      server.close()
+      done()
+    })
+
+    kite.connect()
```
Comment (gokmen) [inline src/server/index.test.js]: 👍

**Comment by gokmen** [comment]
@usirin agreed, let's merge this one in and iterate over there for the initial release. I think we can ship this as is for `1.0.0` to cover Koding requirements for now and we can continue changing with `2.0.0` release. I've added some comments not for the PR only but also as a note to my self, I'll update them and try with Koding then I'll publish it to npm as `1.0.0`, wdyt?

File: src/kite/backoff.js [hunk 6]
```diff
@@ -1,16 +1,22 @@
 const Timeout = require('./timeout')
 
-module.exports = function (options = {}) {
-  const backoff = options.backoff != null ? options.backoff : {}
+module.exports = function(options = {}) {
+  const { backoff = {} } = options.backoff
```
Comment (usirin) [inline src/kite/backoff.js]: I missed this, `s/options.backoff/options`

Comment (gokmen) [inline src/kite/backoff.js]: ah, you are not alone :)

Comment (gokmen) [inline src/kite/backoff.js]: these are addressed in current `master`, fyi

**Pull request merged**